java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -0,0 +1,2943 @@
+/*
+ *
+ *  Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
+/** This class represents the column_view of a column analogous to its cudf cpp counterpart.
+ * It holds view information like the native handle and other metadata for a column_view. It also
+ * exposes APIs that would allow operations on a view.
+ *
+ */
+public class ColumnView implements AutoCloseable, BinaryOperable {
+
+  public static final long UNKNOWN_NULL_COUNT = -1;
+
+  protected long viewHandle = 0;
+  protected DType type;
+  protected long rows;
+  protected Optional<Long> nullCount = Optional.empty();
```

Comment:
nit: all of these feel like they should be final.  I can see viewHandle being non-final because we might want to set it to 0 when we close this, but all of the others are always set.  Even the nullCount, which then I have to ask why we even have it be `Optional`

Reply:
Are u ok with me making nullCounts across host and device column vector not be an optional? I am going to do that now unless u think otherwise

Reply:
I originally made them optional because we wanted to avoid calculating them unless necessary.  It turns out that any time you create a `column_view` from a `column_vector` it will calculate them anyways.  So the only time that it saves you anything would be when we create a `column_view` directly from buffers and we don't know the null count, and happen to never need it.  I can see this potentially changing in the future, but for now  I think dropping the optional aspect of this is fine.

Reply:
I see, I will remove it just from ColumnView for now and go from there if we need to for other pieces, saving us more diffs